Update CO2-brine model documentation (#1330)
GEOSX/GEOSX: src/coreComponents/constitutive/docs/CO2Brine.rst

.. _CO2-EOS:

##################################################################################
CO2-brine model 
##################################################################################

Summary
=======

> The CO2-brine model implemented in GEOSX includes two components (CO2 and H2O) that are transported by one or two fluid phases (the gas (:math:`g`) phase and the brine (:math:`\ell`) phase).

Reviewer: The CO2 can be in supercritical, liquid, or gas phase, right?  maybe we need a better terminology than "gas phase", or add an additional note here?

Author: You are right, I removed the terminology "gas phase" from the documentation and I now use "CO2 phase" everywhere.

Thus, considering the phase component fractions, :math:`y_{c,p}` (i.e., the fraction of the mass of phase :math:`p` represented by component :math:`c`) the following partition matrix determines the component distribution within the two phases:

.. math::
    \begin{bmatrix}
    y_{CO2,g} & y_{CO2,\ell} \\
         0 & 1            \\
    \end{bmatrix}

The update of the fluid properties is done in two steps:

1) The phase fractions (:math:`\nu_p`) and phase component fractions (:math:`y_{c,p}`) are computed as a function of pressure (:math:`p`), temperature (:math:`T`), component fractions (:math:`z_c`), and a constant salinity.

2) The phase densities (:math:`\rho_p`) and phase viscosities (:math:`\mu_p`) are computed as a function of pressure, temperature, the updated phase component fractions, and a constant salinity.

Once the phase fractions, phase component fractions, phase densities, phase viscosities--and their derivatives with respect to pressure, temperature, and component fractions--have been computed, the :ref:`CompositionalMultiphaseFlow` proceeds to the assembly of the accumulation and flux terms.

> Note that the current implementation of the flow solver is isothermal and that the derivatives with respect to temperature are therefore discarded.

Reviewer: @CusiniM should we leave this note here, or will it be outdated soon?  Maybe revise to: "If the solver is isothermal, the derivates w.r.t. temp are discarded"?

Reviewer 2: I think we can leave it for now and I will update it in the thermalCompositionalSolver PR so that the docs gets updated along with the code.

.. math::
   \nu_{\ell} &= \frac{1 + s_{CO2}}{1 + z_{CO2} / ( 1 - z_{CO2} ) } \\
   \nu_{g} &= 1 - \nu_{\ell}

We conclude Step 1 by computing the phase component fractions as:

.. math::
   y_{CO2,\ell} &= \frac{ s_{CO2} }{ 1 + s_{CO2} } \\
   y_{H2O,\ell} &= 1 - y_{CO2,\ell} \\
   y_{CO2,g} &= 1 \\
   y_{H2O,g} &= 0 
    
   
Step 2: Computation of the phase densities and phase viscosities
================================================================

Gas density and viscosity
-------------------------

In GEOSX, the computation of the CO2 gas density and viscosity  is entirely based on look-up in precomputed tables.
The user defines the pressure (in Pascal) and temperature (in degrees Celsius) axis of the density table in the form:

+------------+----------------------+-----------------+-----------------+------------------+-----------------+-----------------+------------------+
| DensityFun | SpanWagnerCO2Density | :math:`p_{min}` | :math:`p_{max}` | :math:`\Delta p` | :math:`T_{min}` | :math:`T_{max}` | :math:`\Delta T` |
+------------+----------------------+-----------------+-----------------+------------------+-----------------+-----------------+------------------+

> This correlation is valid for pressures inferior to :math:`8 \times 10^8` Pascal and temperatures inferior to 800 degrees Celsius.  

Reviewer: ```suggestion
This correlation is valid for pressures less than :math:`8 \times 10^8` Pascal and temperatures less than 800 degrees Celsius.  
```

Author: I fixed the grammar mistakes (this one and the ones below). Thanks for catching them

.. math::
   \frac{p}{RT\rho_{g}} = 1 + \delta \phi^r_{\delta}( \delta, \tau )

where :math:`R` is the gas constant, :math:`\delta := \rho_{g} / \rho_{crit}` is the reduced gas density, and :math:`\tau := T_{crit} / T` is the inverse of the reduced temperature.
The definition of the residual part of the energy equation, denoted by :math:`\phi^r_{\delta}`, can be found in equation (6.5), page 1544 of Span and Wagner (1996).
The coefficients involved in the computation of :math:`\phi^r_{\delta}` are listed in Table (31), page 1544 of Span and Wagner (1996).   
These calculations are done in a preprocessing step.

The pressure and temperature axis of the viscosity table can be parameterized in a similar fashion using the format:

+--------------+----------------------+-----------------+-----------------+------------------+-----------------+-----------------+------------------+
| ViscosityFun | FenghourCO2Viscosity | :math:`p_{min}` | :math:`p_{max}` | :math:`\Delta p` | :math:`T_{min}` | :math:`T_{max}` | :math:`\Delta T` |
+--------------+----------------------+-----------------+-----------------+------------------+-----------------+-----------------+------------------+

> This correlation is valid for pressures inferior to :math:`3 \times 10^8` Pascal and temperatures inferior to 1220 degrees Celsius.  

Reviewer: ```suggestion
This correlation is valid for pressures less than :math:`3 \times 10^8` Pascal and temperatures less than 1220 degrees Celsius.  
```

We currently neglect the critical viscosity, :math:`\mu_{crit}`.
These calculations are done in a preprocessing step.

During the simulation, the update of CO2 gas density and viscosity is simply done with a look-up in the precomputed tables. 

Brine density and viscosity 
---------------------------

The computation of the brine density involves a tabulated correlation presented in Phillips et al (1981). 
The user specifies the (constant) salinity and defines the pressure and temperature axis of the brine density table in the form:

+------------+----------------------+-----------------+-----------------+------------------+-----------------+-----------------+------------------+----------+
| DensityFun | BrineCO2Density      | :math:`p_{min}` | :math:`p_{max}` | :math:`\Delta p` | :math:`T_{min}` | :math:`T_{max}` | :math:`\Delta T` | Salinity | 
+------------+----------------------+-----------------+-----------------+------------------+-----------------+-----------------+------------------+----------+

> The pressure must be in Pascal and must be inferior to :math:`5 \times 10^7` Pascal.

Reviewer: ```suggestion
The pressure must be in Pascal and must be less than :math:`5 \times 10^7` Pascal.
```
